zend_abstract_interface/interceptor/php7/interceptor.c

```diff
@@ -181,18 +203,59 @@ static inline zval *zai_interceptor_get_zval_ptr(const zend_op *opline, int op_t
 #define zai_interceptor_get_zval_ptr_op1(ex) zai_interceptor_get_zval_ptr((ex)->opline, (ex)->opline->op1_type, &(ex)->opline->op1, ex)
 #define zai_interceptor_get_zval_ptr_op2(ex) zai_interceptor_get_zval_ptr((ex)->opline, (ex)->opline->op2_type, &(ex)->opline->op2, ex)
 
+ZEND_TLS zend_op zai_interceptor_custom_return_op;
 static inline void zai_interceptor_return_impl(zend_execute_data *execute_data) {
     zai_interceptor_frame_memory *frame_memory;
     if (zai_hook_memory_table_find(execute_data, &frame_memory)) {
         if (!frame_memory->implicit) {
             zval rv;
             zval *retval = zai_interceptor_get_zval_ptr_op1(execute_data);
-            if (!retval || Z_TYPE_INFO_P(retval) == IS_UNDEF) {
+            bool needs_copy = EX(opline)->op1_type == IS_CONST || EX(opline)->op1_type == IS_CV;
```

Comment:
**question (non-blocking):** @bwoebi just for my understanding, `needs_copy` is true for stuff like
```php
function foo() {
    return 1; // IS_CONST
}
```
or
```php
function foo() {
    $bar = 42;
    return $bar; // IS_CV
}
```
correct?

Reply:
Yes.